File: wakame-plugin/src/main/java/cc/mewcraft/wakame/event/ResourcePackGenerateEvent.java
package cc.mewcraft.wakame.event;

import cc.mewcraft.wakame.pack.generate.GenerationArgs;
import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

/**
 * Called when a resource pack is being generated.
 * <p>
 * You can use this event to change {@link GenerationArgs} to modify the resource pack.
 * <p>
 * To cancel the generation, use {@link #setCancelled(boolean)}.
 */
public class ResourcePackGenerateEvent extends Event implements Cancellable {

    private static final @NotNull HandlerList HANDLERS = new HandlerList();
    private final GenerationArgs args;
    private boolean cancelled;

    public ResourcePackGenerateEvent(GenerationArgs args) {
        super(true);
        this.args = args;
    }

    public GenerationArgs getArgs() {
        return args;
    }

    @Override
    public boolean isCancelled() {
        return cancelled;
    }

    @Override
    public void setCancelled(boolean cancel) {
        cancelled = cancel;
    }

    @Override
    public @NotNull HandlerList getHandlers() {
        return HANDLERS;
    }

    public static @NotNull HandlerList getHandlerList() {
        return HANDLERS;
    }

}

File: wakame-plugin/src/main/kotlin/cc/mewcraft/wakame/event/bukkit/ResourcePackGenerateEvent.kt
package cc.mewcraft.wakame.event.bukkit

import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class ResourcePackGenerateEvent : Event(), Cancellable {
    private var cancel: Boolean = false

    override fun getHandlers(): HandlerList = HANDLER_LIST
    override fun isCancelled(): Boolean {
        return cancel
    }
    override fun setCancelled(cancel: Boolean) {
        this.cancel = cancel
    }

    companion object {
        @JvmStatic
        val HANDLER_LIST = HandlerList()

        @JvmStatic
        fun getHandlerList(): HandlerList = HANDLER_LIST
    }
}